### ISSUE TITLE
Add options for image optimization (other than resizing)

### ISSUE THREAD

**Issue by sebastianbenz** [11ty/eleventy-img#14]
For heavily optimized images, the image size often increases (quite significantly in some cases). A simple fix would be to check the image size and use the original version if it's smaller than the "optimized" one, but it'd be great to include better image compression support by default. 

Has this already been investigated?

**Comment by zachleat** [comment]
Good question! I don’t think we have anything yet for this in place but it would be welcome and most of the pieces already exist

**Comment by zachleat** [comment]
Note that you can specify quality settings for different formats with these options and the output file sizes are included in the results object: https://www.11ty.dev/docs/plugins/image/#advanced-control-of-sharp-image-processor

I wouldn’t consider this resolved _quite_ yet

**Comment by zachleat** [comment]
> This repository is now using [lodash style issue management](https://twitter.com/samselikoff/status/991395669016436736) for enhancements. This means enhancement issues will now be closed instead of leaving them open.

> View [the enhancement backlog here](https://github.com/11ty/eleventy-img/issues?utf8=%E2%9C%93&q=label%3Aneeds-votes+sort%3Areactions-%2B1-desc+). Don’t forget to upvote the top comment with 👍!

**Comment by raffaelj** [comment]
> For heavily optimized images, the image size often increases (quite significantly in some cases). A simple fix would be to check the image size and use the original version if it's smaller than the "optimized" one

I fiddled a bit with using the original image if the processed file is larger...

## Setup

- pretty bare test setup with only 2 dependencies ("@11ty/eleventy": "^3.0.0", "@11ty/eleventy-img": "^6.0.2")
- 18.7 KiB jpg file

With default settings, I have 2 files in my _site folder:

- 23.7 KiB jpeg file
- 22.4 KiB webp file

So in this case both generated files are larger, than the original.

## Expected result:

1. copy original 18.7 KiB jpg file
2. don't create webp file
3. don't create srcset
4. cache result (don't process unneeded variants over and over again)

## What I've tried

I added some quick and dirty code in https://github.com/11ty/eleventy-img/blob/main/src/image.js#L771-L773

```js

            outputFilePromises.push(
              sharpInstance.toFile(stat.outputPath)
                .then(info => {
                  let processedImageIsLarger = info.size > metadata.size;
                  let formatIsLikeOriginal = info.format === metadata.format;
                  if (processedImageIsLarger) {
                    if (formatIsLikeOriginal) {
                      fs.copyFile(this.src, stat.outputPath, (err) => {});
                      stat.size = metadata.size;
                      return stat;
                    }
                    else {
                      fs.unlink(stat.outputPath, (err) => {});
                      return {};
                    }
                  }
                  stat.size = info.size;
                  return stat;
                })
            );
```

I was able to check for the file sizes and copy the original image if larger. If the format doesn't match, I returned an empty object. With this little setup I was able to resolve 1, 2 and 3, but I broke the caching system (4). Now the webp file gets regenerated (and deleted) on every build.

Now I'm stuck (which is fine - I wanted to write content the last hours, but got sidetracked by this image issue instead). Maybe my thoughts and findings are helpful - and now I can revert all my changes in my node_modules folder and finally start writing text instead of code.

**Comment by zachleat** [comment]
This did ship with v6.0.0. Solved by #52.
